middleware/proxy/upstream.go

```diff
@@ -188,7 +203,43 @@ func parseBlock(c *caddyfile.Dispenser, u *staticUpstream) error {
 		u.IgnoredSubDomains = ignoredDomains
 	case "spray":
 		u.Spray = &Spray{}
-
+	case "encoding":
```

Comment:
Maybe instead this should be `protocol` and the default should be `udp`, but we could have a `tcp` option that just does regular DNS using TCP, `tls` for DNS over TLS, and later `https` for rolling in `httpproxy`?